src/gui/qgsadvanceddigitizingdockwidget.cpp

```diff
@@ -870,11 +870,11 @@ bool QgsAdvancedDigitizingDockWidget::applyConstraints( QgsMapMouseEvent *e )
    */
   if ( mLockZButton->isChecked() )
   {
-    point.setZ( mZLineEdit->text().toFloat() );
+    point.setZ( QLocale().toDouble( mZLineEdit->text() ) );
```

Comment:
This should be using QgsAdvancedDigitizingDockWidget::parseUserInput for consistency -- the other x/y/angle/distance widgets accept expression based inputs, so z/m should too

Reply:
It's always used when you press return:
```
  connect( mZLineEdit, &QLineEdit::returnPressed, this, [ = ]() { lockConstraint(); } );
  connect( mMLineEdit, &QLineEdit::returnPressed, this, [ = ]() { lockConstraint(); } );
```